shared/studio/tabs/queryEditor/history.tsx

```diff
@@ -207,6 +207,15 @@ const HistoryItem = observer(function HistoryItem({
           >
             <RelativeTime timestamp={item.timestamp} />
           </div>
+          {state.loadedQueryIndex !== index && (
```

Comment:
I don't think keeping track of the `loadedQueryIndex` is needed. Once you've loaded the history item, it replaces the draft state, so you're essentially no longer viewing that history item. Also there's a couple of bugs:
- Load an item, edit the query, re-open the history. The load button is not available on the last loaded query.
- Loading an item with the 'Enter' key doesn't update the `loadedQueryIndex`.

Reply:
I added `loadedQueryIndex ` because it is was weird to me that user is able to load the same item again if he/she didn't change anything in the draft state. But if we don't wanna keep track of that yes, `loadedQueryIndex ` is not needed. And yes, you are right about 'Enter'. For now I'll remove this state altogether.